src/main/java/com/github/bric3/drain/Drain.java

```diff
@@ -79,6 +93,94 @@ private Drain(int depth,
         this.delimiters = " " + additionalDelimiters;
+        this.clusters = new ArrayList<>();
+        this.root = new Node("(ROOT)", 0);
+    }
+
+    /**
+     * Json creator and Json properties allowing Drain-object import.
+     */
+    @JsonCreator
+    private Drain(@JsonProperty("depth") int depth,
+                  @JsonProperty("similarityThreshold") double similarityThreshold,
+                  @JsonProperty("maxChildPerNode") int maxChildPerNode,
+                  @JsonProperty("delimiters") String additionalDelimiters,
+                  @JsonProperty("clusters") List<LogCluster> clusters,
+                  @JsonProperty("root") Node root) {
+        this.depth = depth;
+        this.similarityThreshold = similarityThreshold;
+        this.maxChildPerNode = maxChildPerNode;
+        this.delimiters = " " + additionalDelimiters;
+        this.clusters = new ArrayList<LogCluster>();
+        this.root = root;
+        logClustersLinker(this.root, this.clusters);
+    }
+
+    /**
+     * Links the list of clusters from tree nodes to Drain.clusters
+     * (needed when importing a model)
+     */
+    private void logClustersLinker(Node aux, List<LogCluster> clusters) {
+        if (aux != null) {
+            for (Node n: aux.allChildren().values()) {
+                if (!n.clusters().isEmpty()) {
+                    clusters.addAll(n.clusters());
+                }
+                logClustersLinker(n, clusters);
+            }
+        }
+    }
+
+    /**
+     * Drain-object exporting functionality which saves a drain model
+     * in a json file at given path.
+     */
+    public void drainExport(String filePath) {
+        // Instance an Object Mapper and allow the access to the Drain object attributes
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
+        
+        // Write the Drain object attributes into path file
+        try {
+            System.out.printf("---- Saving drain model in file %s %n",
+                              filePath);
+            objectMapper.writeValue(new FileOutputStream(filePath), this);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Drain-object importing functionality which creates a Drain instance from the input
+     * json filepath containing Drain object attributes.
+     */
+    public static Drain drainImport(String filePath) throws IOException {
+        // Create an Object mapper
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        // Setup the input .json file
+        File inputDrain = new File(filePath);
+
+        System.out.printf("---- Recovering drain model from file %s %n",
+                          filePath);
+        // Map the attributes from the file and return the Drain instance
+        return objectMapper.readValue(inputDrain, Drain.class);
+    }
+
```

Comment:
I would prefer if this state serialization code remained external to Drain, to keep this class focused on the algorithm itself.